### PR TITLE
Build does not fail anymore, when some service isn't reachable

### PR DESCRIPTION
**Pull request by oniboni**
Improvements:
* Build does not fail anymore, when some service isn't reachable
* In case of fail: adds error admonition + code fence
* Added logging
* UMLet added

### CODE REVIEW

File: kroki/plugin.py [hunk 1]
```diff
@@ -104,7 +114,13 @@ def _download_image(self, matchobj, target, page, files):
         (target / dest_path).mkdir(parents=True, exist_ok=True)
 
         filename = dest_path / f"{ prefix }-{ hash }.svg"
-        urllib.request.urlretrieve(url, target / filename)
+
```
Comment (b-bittner) [inline kroki/plugin.py:117]: Please remove those spaces in line 117

**Review by b-bittner — CHANGES_REQUESTED** [review]
This PR looks great, please remove the spaces at line 117 and also add the underscore (_) in function "on_page_markdown" on line 133, so it is the same as in the other ones.

**Review by fuzzy-focus — CHANGES_REQUESTED** [review]
To be more precise:

`def on_page_markdown(self, markdown, files, page, **kwargs):`
should be
`def on_page_markdown(self, markdown, files, page, **_kwargs):`
for consistency with the rest of the modifications.

File: kroki/plugin.py (hunk 1 above)
Comment (oniboni) [reply kroki/plugin.py:117]:
```suggestion

```

**Review by fuzzy-focus — APPROVED** [review]
👍

**Review by b-bittner — APPROVED** [review]
Looks fantastic, ready for merge...

**Pull request merged**